### PR TITLE
[FEATURE] Added GraphQl Support

### PR DESCRIPTION
**Pull request by lewisvoncken**
Generates:

- GraphQl Schema
- Resolver Class
- Resolver Data Provider Class when Base Type is Query
- Module Dependency when a dependency is injected to the Resolver Data Provider

Base of Magento Core modules which are test cases: StoreGraphQl, QuoteGraphQl, CmsGraphQl, CatalogGraphQl, BundleGraphQl.

image

image

### CODE REVIEW

File: mage2gen/module.py [hunk 1]
```diff
@@ -312,6 +313,167 @@ def save(self, file_path):
...
+# GraphQl Object Type
+###############################################################################
+class GraphQlSchema:
+	template_file = os.path.join(TEMPLATE_DIR, 'graphqlschema.tmpl')
+
+	def __init__(self):
+		self.object_types = []
+
+	def __eq__(self, other):
+		return self
```
Comment (krukas) [inline mage2gen/module.py]: Now there are always equal, is this correct?

Comment (lewisvoncken) [reply mage2gen/module.py]: @krukas  
there can only be one etc/schema.graphqls foreach module so I think so.

Comment (krukas) [reply mage2gen/module.py]: @lewisvoncken Then maybe remove the __eq__ and if a compare is done for merging, remove that and always merge.

Comment (lewisvoncken) [reply mage2gen/module.py]: @krukas thanks for the feedback and this has been removed.

**Pull request merged**